contracts/drafts/issuance/Issuance.sol

```diff
@@ -0,0 +1,210 @@
+     * @param _amount The amount of acceptedToken tokens that will be invested.
+     */
+    function invest(uint256 _amount) external {
+        require(
+            currentState == "OPEN",
+            "Not open for investments."
+        );
+        require(
+            // solium-disable-next-line security/no-block-members
+            now >= openingDate && now <= closingDate,
+            "Not the right time."
+        );
+        require(
+            _amount.mod(issuePrice) == 0,
+            "Fractional investments not allowed."
+        );
+        require(
+            _amount >= minInvestment,
+            "Investment below minimum threshold."
+        );
+
+        acceptedToken.transferFrom(msg.sender, address(this), _amount);
+
+        if (investments[msg.sender] == 0){
+            investors.push(msg.sender);
+        }
+        investments[msg.sender] = investments[msg.sender].add(_amount);
+
+        amountRaised = amountRaised.add(_amount);
+
+        emit InvestmentAdded(msg.sender, _amount);
+    }
+
+    function withdraw() external nonReentrant {
+        require(currentState == "LIVE", "Cannot withdraw now.");
+        require(investments[msg.sender] > 0, "No investments found.");
+        uint256 amount = investments[msg.sender];
+        investments[msg.sender] = 0;
+        issuanceToken.mint(msg.sender, amount.div(issuePrice));
+    }
+
+    /**
+     * @dev Function for an investor to cancel his investment
+     */
+    function cancelInvestment() external nonReentrant {
+        require (
+            currentState == "OPEN" || currentState == "FAILED",
+            "Cannot cancel now."
+        );
+        uint256 amount = investments[msg.sender];
+        investments[msg.sender] = 0;
+        acceptedToken.transfer(msg.sender, amount);
+        emit InvestmentCancelled(msg.sender, amount);
+    }
+
+    /**
+     * @dev Function to open the issuance to investors
+     */
+    function openIssuance() public onlyOwner {
+        require(
+            // solium-disable-next-line security/no-block-members
+            now >= openingDate && now <= closingDate,
+            "Not the right time."
+        );
+        require(
+            address(issuanceToken) != address(0),
+            "Issuance address not set."
+        );
+        transition("OPEN");
+    }
+
+    /**
+     * @dev Function to move to the distributing phase
+     */
+    function startDistribution() public onlyOwner {
+        require(
+            // solium-disable-next-line security/no-block-members
+            now >= closingDate,
+            "Not the right time yet."
+        );
+        require(
+            amountRaised >= softCap,
+            "Not enough funds collected."
+        );
+        transition("LIVE");
+    }
+
+    /**
+     * @dev Function to cancel all investments
+     */
+    function cancelAllInvestments() public onlyOwner{
+        require (
```

Comment:
`transition("FAILED")` already checks we are in a valid state for that transition. You can remove this require.